static/js/topic_data.js

```diff
@@ -55,7 +55,13 @@ exports.topic_history = function () {
             return;
         }
 
-        if (existing.count <= 1) {
+        // The <= in existing.message_id <= message_id covers the
+        // following edge case: If a message with id 933 is removed,
+        // page_params.max_message_id will be will be 932 causing the
+        // local_id of the echoed message to be 932.01. When this new
+        // message with id 934 is deleted, existing.message_id is still
+        // 932 and thus they mismatch.
```

Comment:
`try_deliver_locally` in `echo.js` inserts the local message, the id for which is fetched by `get_next_local_id` which decides the param based on
page_params.max_message_id which is 932 after deleting the message with id 933, making the local id 932.01.